### DISCUSSION TITLE
plot_constraint violations(..)

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#223] [General]
I think, you were right, too complicated:
![image](https://github.com/user-attachments/assets/86d3bfd0-6830-4930-8834-4ee88f0c4d12)
Mabybe better just like this:
_The number of axes depends on the number of instance constraints.
If the user gives two axes, the method will tell how many are needed, unless two is correct_.

If you plan to merge _variable h_ before you change this, maybe it could be added:

_If node_time_interval is a symbol, rather than a float, the times in the instance constraints should be given as
m * node_time_interval, where m is an integer or a float_.

**Comment by moorepants** [comment]
Feel free to improve it in a new PR.

**Reply by Peter230655** [reply]
> Feel free to improve it in a new PR.

The text I see in the API comes directly from the code?
I did not know this.

**Comment by moorepants** [comment]
I still don't like controlling `figsize` inside the function. We didn't do that before and the user simply had to make the figsize long enough (by trial and error) and pass that axes into the function.

**Reply by Peter230655** [reply]
> I still don't like controlling `figsize` inside the function. We didn't do that before and the user simply had to make the figsize long enough (by trial and error) and pass that axes into the function.

I can remove it, of course. My only concern was that if there are many instance constraints (like in the drome example, there are 32), and the user does **not** give axes, the plots are not nice.
Should I remove in in my 'variable h' PR or make a new PR for the version you merged?

**Reply by moorepants** [reply]
>  Should I remove in in my 'variable h' PR or make a new PR for the version you merged?

I don't know what you mean precisely.

**Comment by Peter230655** [comment]
Sorry! What I meant was this:

I made an PR where I changed the plots, but did not touch the variable h printout. You merged this one. So If I make the change here, I will make a new PR

There is a PR pending, not merged, which in addition takes care of the variable h also. Here I'd simply push the changes.

**Comment by Peter230655** [comment]
I think, taken care off.
